goss-testing/scripts/check_ncn_xname.sh

```diff
@@ -0,0 +1,148 @@
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+#
+# This script validates that the following things are true:
+# - The /proc/cmdline file exists and contains the parameter xname=<valid-looking xname>
+# - The file /etc/cray/xname exists and contains that same xname
+#
+# The script will also print the contnets of both files to stdout.
+#
+# In the case that any of the above things are not true, the script will print
+# appropriate error messages to stderr and exit with a 1 return code. If an unexpected
+# error occurs unrelated to the above, the script will print an appropriate error message
+# to stderr and return with a return code other than 0 or 1. Otherwise,
+# it will report success to stdout and exit with a 0 return code.
+#
+
+#
+# NCN xnames should be in the following format:
+# xXcCsSbBnN
+# where:
+# X - 0-9999
+# C - 0-7
+# S - 0-64
+# B - 0-1
+# N - 0-7
+# source: docs-csm: operations/Component_Names_xnames.md
+#
+# grep regular expressions used to match each of these pieces:
+xre="x([0-9]|[1-9][0-9]{0,3})"
+cre="c[0-7]"
+sre="s([0-9]|[1-5][0-9]|6[0-4])"
+bre="b[01]"
+nre="n[0-7]"
+xname_regex="${xre}${cre}${sre}${bre}${nre}"
+
+RC=0
+function err
+{
+    echo "ERROR: $*" 1>&2
+    RC=1
+}
+
+function run_cmd
+{
+    local cmdrc
+    echo "# $*"
+    "$@"
+    cmdrc=$?
+    echo "(return code=$cmdrc)"
+    return $cmdrc
+}
+
+function file_exists_show_contents
+{
+    local funcrc=1
+    if [[ $# -ne 1 ]]; then
+        err "PROGRAMMING LOGIC: $FUNCNAME requires exactly 1 argument, but received $#: $*"
+        exit 2
+    elif [[ -z "$1" ]]; then
+        err "PROGRAMMING LOGIC: Argument to $FUNCNAME may not be blank"
+        exit 3
+    elif [[ ! -e "$1" ]]; then
+        err "File '$1' does not exist"
+    elif ! run_cmd ls -ald "$1" ; then
+        err "Command failed: ls -ald $1"
+    elif [[ ! -f "$1" ]]; then
+        err "File '$1' exists but is not a regular file"
+    elif ! run_cmd cat "$1" ; then
+        err "Command failed: cat $1"
+    else
+        funcrc=0
+    fi
+    return $funcrc
+}
+
+echo "Checking /proc/cmdline"
+if file_exists_show_contents /proc/cmdline ; then
+    # Extract the xname from the file, if it is present
+    if ! PROC_CMDLINE_XNAME_PARAM=$(grep -Eo "[[:space:]]xname=[^[:space:]]+([[:space:]]|$)" /proc/cmdline) ; then
+        err "No xname parameter found in /proc/cmdline"
+    else
+        # We omit the space after the colon because we know the variable will begin with a whitespace
```

Comment:
Is this comment correct (and that same question for line 113)?

Reply:
It should be correct, because our grep command (which is used to set that variable) uses a regular expression that requires it to begin with whitespace.